freedomOfComputing/src/main/java/nl/tudelft/trustchain/foc/community/FOCVoteTracker.kt

```diff
@@ -78,6 +114,20 @@ class FOCVoteTracker(private val activity: MainActivityFOC) {
         return voteMap[fileName]!!.count { v -> v.voteType == voteType }
     }
 
+    /**
+     * Method to take votes from the queue and insert them into the tracker
+     */
+    private suspend fun iterativelyDownloadVotes() {
+        while (scope.isActive) {
+            Log.i("vote-gossip", "${focCommunity.voteMessagesQueue.size} in Queue")
+            while (!focCommunity.voteMessagesQueue.isEmpty()) {
+                val (_, payload) = focCommunity.voteMessagesQueue.remove()
+                insertVote(payload.fileName, payload.focVote)
+            }
+            delay(gossipDelay)
+        }
+    }
+
```

Comment:
Is there a reason why you place the votes in a queue, instead of handling them directly on receive?

Reply:
Because of the way the dependencies are setup currently it has to be done this way. We have created this issue #36 to try and improve this in the future